includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -490,6 +496,51 @@ protected function get_row_data( $item ) {
 		return $escaped_column_data;
 	}
 
+	/**
+	 * Get the date on which the last lesson was marked complete in a course by any student.
+	 *
+	 * @since 4.2.0
+	 *
+	 * @param int $course_id Course ID.
+	 * @return string Date on which the last lesson was marked complete in a course by any student, or N/A if none.
+	 */
+	private function get_course_last_activity_date( int $course_id ): string {
+		$last_activity = Sensei_Utils::sensei_check_for_activity(
+			[
+				'post__in' => Sensei()->course->course_lessons( $course_id, 'any', 'ids' ),
+				'type'     => 'sensei_lesson_status',
+				'status'   => [ 'complete', 'passed', 'graded' ],
+				'number'   => 1,
+			],
+			true
+		);
+
+		if ( ! $last_activity ) {
+			return 'N/A';
+		}
+
+		// Return the full date when doing a CSV export.
+		if ( $this->csv_output ) {
+			return $last_activity->comment_date_gmt;
+		}
+
+		$timezone           = new DateTimeZone( 'GMT' );
+		$now                = new DateTime( 'now', $timezone );
+		$last_activity_date = new DateTime( $last_activity->comment_date_gmt, $timezone );
+		$diff_in_days       = $now->diff( $last_activity_date )->days;
+
+		// Show a human readable date if activity is within 6 days.
+		if ( $diff_in_days < 7 ) {
+			return sprintf(
+				/* translators: Time difference between two dates. %s: Number of seconds/minutes/etc. */
+				__( '%s ago', 'sensei-lms' ),
+				human_time_diff( strtotime( $last_activity->comment_date_gmt ) )
+			);
+		}
+
+		return wp_date( get_option( 'date_format' ), $last_activity->getTimestamp(), $timezone );
```

Comment:
Hm, suspiciously familiar. 😄 

There is some code duplication with [`get_last_activity`](https://github.com/Automattic/sensei/blob/feature/reports/includes/class-sensei-analysis-overview-list-table.php#L656-L679). What do you think about [drying](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) the code a bit? Maybe make a method that accepts the last activity comment and returns the time string?

Also, is this important enough to be covered by tests? I guess some of the logic is already covered by the tests in https://github.com/Automattic/sensei/pull/4863 but the tests may need some refactoring if the logic is moved to a new method.

Reply:
Now I see that this is not ready for review... 😅